pkg/miniogw/gateway-storj.go

```diff
@@ -5,31 +5,143 @@ package storj
 
 import (
 	"context"
+	"crypto/sha256"
+	"flag"
+	"fmt"
 	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
 	"time"
 
 	"github.com/minio/cli"
-
-	"github.com/minio/minio/cmd"
+	minio "github.com/minio/minio/cmd"
 	"github.com/minio/minio/pkg/auth"
 	"github.com/minio/minio/pkg/hash"
+	"github.com/vivint/infectious"
+
+	"storj.io/storj/pkg/eestream"
+)
+
+var (
+	pieceBlockSize = flag.Int("piece_block_size", 4*1024, "block size of pieces")
+	key            = flag.String("key", "a key", "the secret key")
+	rsk            = flag.Int("required", 20, "rs required")
+	rsn            = flag.Int("total", 40, "rs total")
 )
 
 func init() {
-	cmd.RegisterGatewayCommand(cli.Command{
+	minio.RegisterGatewayCommand(cli.Command{
 		Name:            "storj",
 		Usage:           "Storj",
 		Action:          storjGatewayMain,
 		HideHelpCommand: true,
 	})
 }
 
+// getbucketscallback returns the buckets list
+func (s *storjObjects) getBuckets() (buckets []minio.BucketInfo, err error) {
```

Comment:
This method does not seems to have a case where an error is returned. Thus, why having and error as a return value?

Reply:
Same for uploadFile() and getFiles()

Reply:
I have it for future development when we start integrating with high client, you will need to have the error conditions reported and handled. No change